#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
UVC Camera Driver dynamic_reconfigure 参数定义
"""

PACKAGE = "uvc_camera_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 触发模式
trigger_enum = gen.enum([
    gen.const("VideoStream", int_t, 0, "正常视频流模式"),
    gen.const("ExternalTrigger", int_t, 1, "外触发模式")
], "触发模式选项")
gen.add("trigger_mode", int_t, 0, "触发模式 (0=视频流, 1=外触发)", 0, 0, 1, edit_method=trigger_enum)

# 曝光控制
gen.add("exposure_auto", bool_t, 0, "自动曝光", False)
gen.add("exposure_absolute", int_t, 0, "曝光时间 (单位: 100us)", 20, 1, 10000)

# 白平衡控制
gen.add("white_balance_auto", bool_t, 0, "自动白平衡", False)
gen.add("white_balance_temperature", int_t, 0, "色温", 4600, 2800, 6500)

# 图像调整
gen.add("brightness", int_t, 0, "亮度", 0, -64, 64)
gen.add("contrast", int_t, 0, "对比度", 50, 0, 100)
gen.add("saturation", int_t, 0, "饱和度", 50, 0, 100)
gen.add("hue", int_t, 0, "色调", 0, -180, 180)
gen.add("gamma", int_t, 0, "伽马值", 348, 100, 500)
gen.add("sharpness", int_t, 0, "锐度", 50, 0, 100)
gen.add("backlight_compensation", int_t, 0, "背光补偿", 0, 0, 2)

# 电源频率
power_line_enum = gen.enum([
    gen.const("Disabled", int_t, 0, "禁用"),
    gen.const("Hz50", int_t, 1, "50Hz"),
    gen.const("Hz60", int_t, 2, "60Hz")
], "电源频率选项")
gen.add("power_line_frequency", int_t, 0, "电源频率抗闪烁", 0, 0, 2, edit_method=power_line_enum)

# 时间偏移 (用于校准USB相机触发延迟)
gen.add("time_offset", double_t, 0, "时间偏移量 (秒), 用于校准触发延迟", -0.007, -1.0, 1.0)

exit(gen.generate(PACKAGE, "uvc_camera_driver", "CameraParams"))
